lib/elixir/test/elixir/kernel/with_test.exs

```diff
@@ -47,6 +47,15 @@ defmodule Kernel.WithTest do
     assert result == :error
   end
 
+  test "doesn't not leak variables to else" do
+    state = 1
+    result =
+      with 1 <- state, state = 2,
+           :ok <- error(), do: state, else: (_ -> state)
```

Comment:
Maybe we can have this `with` on one line so we don't have the whitespace pyramid?